services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/core/settings.py

```diff
@@ -31,6 +31,13 @@ class ApplicationSettings(BaseCustomSettings, MixinLoggingSettings):
         "Sidecar must have r/w permissions in this folder.",
     )
 
+    DYNAMIC_SIDECAR_SHARED_STORE_DIR: Path = Field(
+        ...,
+        description="Directory where the dynamic-sidecar persists "
```

Comment:
sorry I'm a bit confused here. What do you call a SharedStore? is that some state of the dynamic-sidecar that gets stored there?

Reply:
Yes, that is actually correct. What is called `SharedStore` is actually what keeps track of the submitted docker-compose spec and of the started containers names.
As usual I am opened to better naming, since this one appears not to be helpful at all.

Reply:
why is it shared actually? why do you call it shared?
It's a StateStorage or RuntimeStateStorage...

Reply:
Name is very old, no longer has something to do with it's original. It's just a storage area where information relative to the application's state which need to be persisted are kept.
What about calling it `PersistantState`? Does that give a better idea of what it's used for? @pcrespov @sanderegg

Reply:
actually wondering if you should not just dump the AppState in a specific folder (could be the same for any of our apps) and then reload if it if it exists on start.

Reply:
Not seeing a benefit in doing so. Usually you have database connection. Objects keeping track of tasks and some information regarding the current state.
You'd have to filter out most of them out before dumping it. If I were to add something like that I'd make a include filter to pull in properties I'd like to include from the state, when dumping.